frontend/public/components/configmap-and-secret-data.tsx

```diff
@@ -99,56 +99,77 @@ export const SecretValue: React.FC<SecretValueProps> = ({ value, reveal, encoded
 };
 SecretValue.displayName = 'SecretValue';
 
-export const SecretData: React.FC<SecretDataProps> = ({ data, title }) => {
+const SecretDataRevealButton: React.FC<SecretDataRevealButtonProps> = ({ reveal, onClick }) => {
+  const { t } = useTranslation();
+  return (
+    <Button
+      type="button"
+      onClick={onClick}
+      variant="link"
+      className="pf-m-link--align-right"
+      data-test="reveal-values"
+    >
+      {reveal ? (
+        <>
+          <EyeSlashIcon className="co-icon-space-r" />
+          {t('public~Hide values')}
+        </>
+      ) : (
+        <>
+          <EyeIcon className="co-icon-space-r" />
+          {t('public~Reveal values')}
+        </>
+      )}
+    </Button>
+  );
+};
+
+export const SecretData: React.FC<SecretDataProps> = ({ data }) => {
   const [reveal, setReveal] = React.useState(false);
+  const [hasRevealableContent, setHasRevealableContent] = React.useState(false);
   const { t } = useTranslation();
-  const titleI18n = title || t('public~Data');
-  const dl = [];
-  Object.keys(data || {})
-    .sort()
-    .forEach((k) => {
-      dl.push(
-        <dt i18n-not-translated="true" key={`${k}-k`} data-test="secret-data-term">
-          {k}
-        </dt>,
-      );
-      dl.push(
-        <dd key={`${k}-v`}>
-          {ITOB.isBinary(k, Buffer.from(data[k], 'base64')) ? (
-            <DownloadBinaryButton label={k} value={data[k]} />
-          ) : (
-            <SecretValue value={data[k]} reveal={reveal} id={k} />
-          )}
-        </dd>,
-      );
-    });
+
+  const dataDescriptionList = React.useMemo(() => {
+    return data
+      ? Object.keys(data)
+          .sort()
+          .map((k) => {
+            const isBinary = ITOB.isBinary(k, Buffer.from(data[k], 'base64'));
+            if (!isBinary) {
+              if (data[k]) {
```

Comment:
During my testing of the changes addressing your review @cyril-ui-developer, I stumbled upon a problem where secrets data for existing key may have no value and in this case we were still showing the Reveal button ( see screenshots ). This is out of scope of the original issue, but since the button has no effect when revealing/hiding these kind of secrets, we should address it as well. This condition accomplishes that we only set the state variable `hasRevealableContent` to true only when there is a non-binary file that also has a value.

<img width="1224" alt="Screenshot 2024-11-13 at 10 25 26" src="https://github.com/user-attachments/assets/995c10a3-9de4-480b-9f93-c48ca06b19d0">

<img width="1220" alt="Screenshot 2024-11-13 at 10 25 39" src="https://github.com/user-attachments/assets/816fb7b5-75f4-4523-b793-9e840f57e7ab">

Reply:
> During my testing of the changes addressing your review @cyril-ui-developer, I stumbled upon a problem where secrets data for existing key may have no value and in this case we were still showing the Reveal button ( see screenshots ). This is out of scope of the original issue, but since the button has no effect when revealing/hiding these kind of secrets, we should address it as well. This condition accomplishes that we only set the state variable `hasRevealableContent` to true only when there is a non-binary file that also has a value.
> 
> <img alt="Screenshot 2024-11-13 at 10 25 26" width="1224" src="https://private-user-images.githubusercontent.com/49416557/385705127-995c10a3-9de4-480b-9f93-c48ca06b19d0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MTc0OTcsIm5iZiI6MTczMTUxNzE5NywicGF0aCI6Ii80OTQxNjU1Ny8zODU3MDUxMjctOTk1YzEwYTMtOWRlNC00ODBiLTlmOTMtYzQ4Y2EwNmIxOWQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDE2NTk1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwMjU1OGNiYjZlYTEzMzQzZWMwZTgxYzk0ZjU5OGJiMWJjM2Q0NzBkMzA2ODc5YjM5OTk0OThmYmIyYmQ0OGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AXW7LOtlqeicWxqxr2aeKRrEpT0zvl5x7xBoqjELbvQ"> <img alt="Screenshot 2024-11-13 at 10 25 39" width="1220" src="https://private-user-images.githubusercontent.com/49416557/385705188-816fb7b5-75f4-4523-b793-9e840f57e7ab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MTc0OTcsIm5iZiI6MTczMTUxNzE5NywicGF0aCI6Ii80OTQxNjU1Ny8zODU3MDUxODgtODE2ZmI3YjUtNzVmNC00NTIzLWI3OTMtOWU4NDBmNTdlN2FiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDE2NTk1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUwZjllZjVlY2YxZjYxMzUxYjU3YjI1OTBiMGFhNWZhZmFlMDAzZmE1YjlhODZjMzEyZjM1MzMwZGJiNTQ5MDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KCT4P7cspFx6iP8cfdb7ftkR2dV9sp39zjKzbVXgKME">

@Mylanos Good catch and nice work. :)
nit: Better to use one `if` statement here and combine both conditions with `&&` operator.